run_dir/design/flowcells.html

```diff
@@ -17,63 +17,136 @@ <h1><span id="page_title">Flowcells</span> </h1>
-                {% else%}
-                    -
-                {% end %}
+                <td class="run_mode">{{ flowcells[onefc].get('run_mode', '-') }}</td>
+                <td class="recipe">{{ flowcells[onefc].get('recipe', '-') }}</td>
+                <td class="lane" class="text-center">
+                     {% if 'lane_info' in flowcells[onefc] > 0 %}
+                         {% for onelane in flowcells[onefc]['lane_info'] %}
+                             <samp style="max-width:50px;">{{ onelane }}</samp><br>
+                         {% end %}
+                    {% end %}
                 </td>
-                <td class="mode">{{ flowcells[onefc].get('mode', '-') }}</td>
-                <td class="chemistry" class="text-center">
-                {% if 'fctype' in flowcells[onefc] %}
-                    {{ flowcells[onefc]['fctype'].split(" ")[-1] }}
-                {% elif 'kitver' in flowcells[onefc] %}
-                    v{{ flowcells[onefc]['kitver'][-1:] }}
-                {% elif 'workflow' in flowcells[onefc] %}
-                    {{ flowcells[onefc]['workflow'].replace("NovaSeq", "") }}
-                {% else %}
-                -
-                {% end %}
+                <td class="project" class="text-center">
+                     {% if 'lane_info' in flowcells[onefc] > 0 %}
+                         {% for onelane in flowcells[onefc]['lane_info'] %}
+                             <samp style="max-width:50px;">
+                             {% for oneproj in flowcells[onefc]['lane_info'][onelane]['project_name'] %}
+                                 <a href="/project/{{ flowcells[onefc]['lane_info'][onelane]['project_id'][flowcells[onefc]['lane_info'][onelane]['project_name'].index(oneproj)] }}">{{ oneproj }}</a>
+                                 {% if flowcells[onefc]['lane_info'][onelane]['project_name'].index(oneproj) < len(flowcells[onefc]['lane_info'][onelane]['project_name'])-1 %}
+                                     {{ ";" }}
+                                 {% end %}
+                             {% end %}
+                             </samp><br>
+                         {% end %}
+                    {% end %}
+                </td>
+                <td class="pf_clusters" class="text-center">
+                     {% if 'lane_info' in flowcells[onefc] > 0 %}
+                         {% for onelane in flowcells[onefc]['lane_info'] %}
+                             {% if flowcells[onefc]['lane_info'][onelane].get('pf_clusters') %}
+                                 <samp style="max-width:50px;">{{ int(flowcells[onefc]['lane_info'][onelane].get('pf_clusters').replace(',',''))/1000000 }}</samp>
+                                 {% set threshold = thresholds.get(flowcells[onefc].get('run_mode', ''), 0) %}
+                                 {% if int(flowcells[onefc]['lane_info'][onelane].get('pf_clusters').replace(',',''))/1000000 >= threshold %}
```

Comment:
Hi, for the Q30 part I actually used the codes in `flowcell.py`. I totally agree that it would be better to have a common QC criteria function with everything defined separately.